tests/indexes/tests.py

```diff
@@ -21,7 +21,24 @@ def test_index_name_hash(self):
                 column_names=("c1", "c2", "c3"),
                 suffix="123",
             )
-        self.assertEqual(index_name, "indexes_article_c1_7ce4cc86123")
+        self.assertEqual(index_name, "indexes_article_c1_c2_c3_7ce4cc86123")
+
+    def test_index_name_trucation(self):
+        # Test truncation on various databases.
+        long_name = 'l%sng' % ('o' * 86)
+        with connection.schema_editor() as editor:
+            index_name = editor._create_index_name(
+                model=Article,
+                column_names=(long_name, 'c1', 'c2'),
+                suffix='ix',
+            )
+        expected = {
+            'mysql': 'indexes_article_looooooooooooooooooooooooo_2ccf1189ix',
+            'oracle': 'indexes_a_loooooooo_2ccf1189ix',
+            'postgresql': 'indexes_article_loooooooooooooooooooooooo_2ccf1189ix',
+            'sqlite': 'indexes_article_l%sng_c1_c_2ccf1189ix' % ('o' * 86),
+        }
+        self.assertEqual(index_name, expected[connection.vendor])
```

Comment:
This test isn't extensible for third-party database backends. I'm not sure about the best approach. Any input @manfre (maintainer of django-mssql)?